Sources/Sampler/metropolis_local_v2.hpp

```diff
@@ -0,0 +1,154 @@
+// Copyright 2018 The Simons Foundation, Inc. - All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef SOURCES_SAMPLER_METROPOLIS_LOCAL_V2_HPP
+#define SOURCES_SAMPLER_METROPOLIS_LOCAL_V2_HPP
+
+#include <functional>
+#include <limits>
+#include <vector>
+
+#include <Eigen/Core>
+#include <nonstd/span.hpp>
+
+#include "Machine/rbm_spin_v2.hpp"
+#include "Operator/abstract_operator.hpp"
+#include "Utils/exceptions.hpp"
+#include "Utils/random_utils.hpp"
+
+namespace netket {
+
+struct Suggestion {
```

Comment:
I think this is more commonly called `Proposal` (or proposed update) in MC literature.

Reply:
I think that if we go through this route, we need to write a general markov chain monte carlo abstract class, and implement the other cases as derived ones. Doing this change only here has the effect of making the code too complex without achieving generality